### PR TITLE
feat: show organization name in workspaces table

### LINKED ISSUE #14278: :star: Add organization context to the workspaces table
Add organization information to the workspaces table. One option is to put the organization name underneath the Template name.

**Requirements**

- [x] add org name only (not a link) to workspace table
- [x] only display org name according to existing `useDashboard` hook's `dashboard.showOrganizations` prop - if requirements shift, this ticket is no longer valid
- [ ] add test for conditional logic 
- [ ] timebox design to 1 day
-

Screenshot from 2024-08-15 13-55-52

### PR DESCRIPTION
**Pull request by buenos-nachos**
Closes #14278

## Changes made
- Added logic to show organization name when the `showOrganization` property from `useDashboard` is `true`
- Added text labels for username and organization name to help remove UI ambiguity
- Added small Storybook test to make sure that organization info is accessible
- Made a slight tweak to the CSS for the `AvatarData` component to make sure that its line height can't inherently incorrectly

Screenshot 2024-09-03 at 3 49 25 PM

### CODE REVIEW

File: site/src/pages/WorkspacesPage/WorkspacesPageView.stories.tsx [hunk 1]
```diff
@@ -265,3 +266,30 @@ export const InvalidPageNumber: Story = {
...
+		const canvas = within(canvasElement);
+		const accessibleTableCell = await canvas.findByRole("cell", {
+			// Need whitespace classes because different parts of the element
+			// are going in different spans, and Storybook doesn't consolidate
+			// these
+			name: /org\s?(?:anization)?\s?:\s?Limbus Co\./i,
+		});
+
+		expect(accessibleTableCell).toBeDefined();
+	},
```
Comment (Parkreiner) [inline site/src/pages/WorkspacesPage/WorkspacesPageView.stories.tsx:294]: Just wanted to make sure that I understand the limitations around decorators: they're basically assumed to always be static values, right? There's no easy way to make a decorator re-render with a new value?

Comment (aslilac) [reply site/src/pages/WorkspacesPage/WorkspacesPageView.stories.tsx:294]: each "decorator" undergoes a small transformation to get turned into a component by storybook. there's no way to force a rerender from the outside, just like any other component. being a component tho, they can call hooks and have state, which is kind of non-obvious.

File: site/src/pages/WorkspacesPage/WorkspacesTable.tsx [hunk 2]
```diff
@@ -189,7 +192,29 @@ export const WorkspacesTable: FC<WorkspacesTableProps> = ({
...
+												lineHeight: 1.5,
+											}}
+										>
+											{/*
+												Only using shorthand version of "organizations" for aesthetics,
+												but because screen readers don't care about aesthetics, we can
+												always display the full text to them
+											*/}
+											org
+											<span css={{ ...visuallyHidden }}>anization</span>:
```
Comment (aslilac) [inline site/src/pages/WorkspacesPage/WorkspacesTable.tsx:214]: are screen readers going to handle this split well though? my gut feeling is no

File: site/src/components/AvatarData/AvatarData.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ export const AvatarData: FC<AvatarDataProps> = ({
 						css={{
 							fontSize: 13,
 							color: theme.palette.text.secondary,
-							lineHeight: "150%",
+							lineHeight: 1.5,
```
Comment (aslilac) [inline site/src/components/AvatarData/AvatarData.tsx:57]: ah the wonderful subtleties of css. so annoying that this actually means something different. 🙃

**Review by aslilac — APPROVED** [review]
only thing is that I'm wondering if we really even need the labels. curious what the rationale was.

**Comment by Parkreiner** [comment]
@aslilac Sorry I didn't get back to this yesterday, but now that I think about it, I think you're right, and that the labels are unnecessary

I guess my worry was that, as we keep cramming multiple values into single table cells, we still need a way to structure the information. Because the HTML spec basically assumes that table cells will basically be like database columns – no more than one value per cell. So anytime we diverge from that, we can't assume that we'll have accessible info just because we're using the correct baseline semantics. I think I'm also just a bit skittish about multiple unlabeled values in general

I still think the labels are good for screen readers, but I do think it's overkill for sighted users. Let me switch things around, which should also make it easier to maintain on our end

File: site/src/components/AvatarData/AvatarData.tsx (hunk 3 above)
Comment (Parkreiner) [reply site/src/components/AvatarData/AvatarData.tsx:57]: CSS is so silly sometimes

File: site/src/pages/WorkspacesPage/WorkspacesTable.tsx (hunk 2 above)
Comment (Parkreiner) [reply site/src/pages/WorkspacesPage/WorkspacesTable.tsx:214]: I tried it with the built-in Mac screen reader, but (1) I think I was just assuming that any elements that are semantically inline elements wouldn't affect pronunciation, and (2) the Mac screen reader isn't even the most popular reader by a long shot

Let me change this just to be on the safe side

**Pull request merged**